arangod/Transaction/Methods.cpp

```diff
@@ -2640,69 +2667,63 @@ OperationResult transaction::Methods::truncateLocal(
   }
 
   // Now see whether or not we have to do synchronous replication:
-  if (_state->isDBServer()) {
-    // Now replicate the same operation on all followers:
-    auto const& followerInfo = collection->followers();
-    std::shared_ptr<std::vector<ServerID> const> followers = followerInfo->get();
+  if (replicationType == ReplicationType::LEADER) {
+    TRI_ASSERT(followers != nullptr);
 
-    if (!isFollower && followers->size() > 0) {
-      // Now replicate the good operations on all followers:
-      auto cc = arangodb::ClusterComm::instance();
+    // Now replicate the good operations on all followers:
+    auto cc = arangodb::ClusterComm::instance();
 
-      if (cc != nullptr) {
-        // nullptr only happens on controlled shutdown
-        std::string path =
-            "/_db/" +
-            arangodb::basics::StringUtils::urlEncode(vocbase().name()) +
-            "/_api/collection/" +
-            arangodb::basics::StringUtils::urlEncode(collectionName) +
-            "/truncate?isSynchronousReplication=" +
-            ServerState::instance()->getId();
-        auto body = std::make_shared<std::string>();
+    if (cc != nullptr) {
```

Comment:
isn't this just happening in the case of a shutdown ?

Reply:
yes. But we must not call something on the nullptr, so we need the check.

Reply:
why not ```if (cc == nullptr) return TRI_ERROR_SHUTTING_DOWN```

Reply:
there is some code running after the code, so we cannot retun from here